symfony/plugins/orangehrmRESTPlugin/lib/Api/User/EmployeePunchInAPI.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+/**
+ * OrangeHRM is a comprehensive Human Resource Management (HRM) System that captures
+ * all the essential functionalities required for any enterprise.
+ * Copyright (C) 2006 OrangeHRM Inc., http://www.orangehrm.com
+ *
+ * OrangeHRM is free software; you can redistribute it and/or modify it under the terms of
+ * the GNU General Public License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * OrangeHRM is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+
+namespace Orangehrm\Rest\Api\User;
+
+use Orangehrm\Rest\Api\Exception\InvalidParamException;
+use Orangehrm\Rest\Api\Exception\RecordNotFoundException;
+use Orangehrm\Rest\Api\Exception\BadRequestException;
+use Orangehrm\Rest\Http\Response;
+use Orangehrm\Rest\Api\Attendance\PunchInAPI;
+use \PluginAttendanceRecord;
+use \sfContext;
+use \DateTimeZone;
+use \DateTime;
+use \AttendanceRecord;
+
+
+class EmployeePunchInAPI extends PunchInAPI
+{
+    /**
+     * @return Response
+     * @throws InvalidParamException
+     * @throws RecordNotFoundException
+     */
+    public function savePunchIn()
+    {
+        $filters = $this->filterParameters();
```

Comment:
Please reconsider the variable name.

Filters are there to filter out a result set. $filters variable is not being used to do any filtering, rather hold parameters to initiate/update an Attendance Record.